src/Hamiltonians/ExtendedHubbardReal1D.jl

```diff
@@ -37,24 +62,26 @@ end
 dimension(::ExtendedHubbardReal1D, address) = number_conserving_dimension(address)
 
 LOStructure(::Type{<:ExtendedHubbardReal1D{<:Real}}) = IsHermitian()
+LOStructure(::Type{<:ExtendedHubbardReal1D{<:Complex}}) = IsHermitian()
 
 Base.getproperty(h::ExtendedHubbardReal1D, s::Symbol) = getproperty(h, Val(s))
 Base.getproperty(h::ExtendedHubbardReal1D, ::Val{:address}) = getfield(h, :address)
 Base.getproperty(h::ExtendedHubbardReal1D{<:Any,<:Any,U}, ::Val{:u}) where U = U
 Base.getproperty(h::ExtendedHubbardReal1D{<:Any,<:Any,<:Any,V}, ::Val{:v}) where V = V
 Base.getproperty(h::ExtendedHubbardReal1D{<:Any,<:Any,<:Any,<:Any,T}, ::Val{:t}) where T = T
+Base.getproperty(h::ExtendedHubbardReal1D{<:Any,<:Any,<:Any,<:Any,<:Any,BOUNDARY_CONDITION}, ::Val{:boundary_condition}) where BOUNDARY_CONDITION = BOUNDARY_CONDITION
 
 function num_offdiagonals(::ExtendedHubbardReal1D, address::SingleComponentFockAddress)
     return 2 * num_occupied_modes(address)
 end
 
 """
-    extended_bose_hubbard_interaction(address, m)
+    extended_hubbard_interaction(address, m)
```

Comment:
This whole docstring is a bit messed up (function signature is wrong, formulas don't render correctly). I suggest something like this:

```suggestion
    extended_hubbard_interaction(h::ExtendedHubbardReal1D, address)
    
Compute and return both the nearest neighbor occupation number product ``\\sum_j n_j n_{j+1}`` 
(according to the boundary conditions of `h`) as well as the on-site product ``\\sum_j n_j (n_j - 1)`` 
treating the `address` as a one-dimensional chain.

See [`ExtendedHubbardReal1D`](@ref) and [`hopnextneighbour`](@ref).
```